2-api/server.js

```diff
@@ -0,0 +1,227 @@
+
+        pool
+          .query(query, [newProduct])
+          .then(() => res.send("New product added"))
+          .catch((e) => console.error(e));
+      }
+    });
+});
+
+app.post("/availability", function (req, res) {
+  const productId = req.body.prod_id;
+  const supplierId = req.body.supp_id;
+  const unitPrice = req.body.unit_price;
+
+  if (unitPrice <= 0) {
+    return res.status(400).send("Invalid price");
+  }
+
+  pool
+    .query(
+      "SELECT p.id, s.id FROM  products p, suppliers s  WHERE p.id=$1 AND s.id=$2",
+      [productId, supplierId]
+    )
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400).send("Product or Supplier is not Exists");
+      } else {
+        const query =
+          "INSERT INTO product_availability (prod_id, supp_id, unit_price) VALUES ($1, $2, $3)";
+
+        pool
+          .query(query, [productId, supplierId, unitPrice])
+          .then(() => res.send("New product availability added"))
+          .catch((e) => console.error(e));
+      }
+    });
+});
+
+app.post("/customers/:customerId/orders", function (req, res) {
+  const customerId = req.params.customerId;
+  const orderDate = req.body.order_date;
+  const orderRef = req.body.order_reference;
+
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400).send("Customer not Exist");
+      } else {
+        const query =
+          "INSERT INTO orders (order_date, order_reference, customer_id)VALUES ($1, $2, $3)";
+        pool
+          .query(query, [orderDate, orderRef, customerId])
+          .then(() => res.send("New order created"))
+          .catch((e) => console.error(e));
+      }
+    });
+});
+
+app.put("/customers/:customerId", function (req, res) {
+  const customerId = req.params.customerId;
+  const newName = req.body.name;
+  const newAdd = req.body.address;
+  const newCity = req.body.city;
+  const newCountry = req.body.country;
+
+  pool
+    .query("SELECT * FROM customers WHERE id=$1", [customerId])
+    .then((result) => {
+      if (result.rowCount === 0) {
+        return res.status(400).send("Customer not exist");
+      } else {
+        const query =
+          "UPDATE customers SET name=$1, address=$2, city=$3, country=$4 WHERE id=$5";
+
+        pool
+          .query(query, [newName, newAdd, newCity, newCountry, customerId])
+          .then(() => res.send("Customer info been update"))
+          .catch((e) => console.error(e));
+      }
+    });
+});
+
+app.delete("/orders/:orderId", function (req, res) {
+  const orderId = req.params.orderId;
+
+  pool
+    .query("DELETE FROM orders WHERE customer_id=$1", [orderId])
+    .then(() => {
+      pool
+        .query("DELETE FROM customers WHERE id=$1", [orderId])
+        .then(() => res.send("customer deleted"))
+        .catch((e) => console.error(e));
```

Comment:
You need to delete from the orders table where the id is the provided id and from the order items table where the order id is the provided id.